### PR TITLE
PDA Pen Holder

### PR DESCRIPTION
**Pull request by exp111**

### CODE REVIEW

File: Content.Server/GameObjects/Components/PDA/PDAComponent.cs [hunk 1]
```diff
@@ -40,12 +41,18 @@ public class PDAComponent : SharedPDAComponent, IInteractUsing, IActivate, IUse,
         [Dependency] private readonly IEntityManager _entityManager = default!;
 
         [ViewVariables] private Container _idSlot = default!;
+        [ViewVariables] private Container _penSlot = default!;
+
         [ViewVariables] private bool _lightOn;
+
         [ViewVariables] private string _startingIdCard = default!;
-        [ViewVariables] public bool IdSlotEmpty => _idSlot.ContainedEntities.Count < 1;
+        [ViewVariables] private string _startingPen = default!;
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/Components/PDA/PDAComponent.cs:49]: Should be nullable

File: Content.Server/GameObjects/Components/PDA/PDAComponent.cs [hunk 2]
```diff
@@ -183,11 +199,62 @@ public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
...
+            {
+                return true;
+            }
+
+            // Insert Pen
+            _penSlot.Insert(item);
+
+            if (swap != null)
+            {
+                eventArgs.User.GetComponent<HandsComponent>().PutInHand(swap.GetComponent<ItemComponent>());
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/Components/PDA/PDAComponent.cs:235]: 50/50 on it being dependent upon hands but it's fine for now.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/GameObjects/Components/PDA/PDAComponent.cs [hunk 3]
```diff
@@ -40,12 +41,18 @@ public class PDAComponent : SharedPDAComponent, IInteractUsing, IActivate, IUse,
         [Dependency] private readonly IEntityManager _entityManager = default!;
 
         [ViewVariables] private Container _idSlot = default!;
+        [ViewVariables] private Container _penSlot = default!;
```
[the remark this answers is not in the document]
Comment (exp111) [reply Content.Server/GameObjects/Components/PDA/PDAComponent.cs:44]: Yes the problem is that the swap code is kinda shit.

**Review by metalgearsloth — APPROVED** [review]
LGTM though having a button for it in the UI would be nice like the ID (either a TODO or implementing it is fine).

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**